File: src/MagnusAddon/viewmodels/kanji_list/sentence_kanji_list_viewmodel.py
from ankiutils import app
from viewmodels.kanji_list.kanji_list_viewmodel import KanjiListViewModel
from viewmodels.kanji_list.sentence_kanji_viewmodel import KanjiViewModel


def create(kanji: list[str]) -> KanjiListViewModel:
    kanji_notes = app.col().kanji.with_any_kanji(kanji)
    kanji_viewmodels = [KanjiViewModel(note) for note in kanji_notes]
    return KanjiListViewModel(kanji_viewmodels)

File: src/src_dotnet/JAStudio.Core/ViewModels/KanjiList/SentenceKanjiListViewModel.cs
using System.Collections.Generic;
using System.Linq;

namespace JAStudio.Core.ViewModels.KanjiList;

public class SentenceKanjiListViewModel
{
   readonly TemporaryServiceCollection _services;
   internal SentenceKanjiListViewModel(TemporaryServiceCollection services) => _services = services;

    public KanjiListViewModel Create(List<string> kanji)
    {
        var kanjiNotes = TemporaryServiceCollection.Instance.App.Col().Kanji.WithAnyKanjiIn(kanji);
        var kanjiViewModels = kanjiNotes.Select(note => new KanjiViewModel(note)).ToList();
        return new KanjiListViewModel(kanjiViewModels);
    }
}
